containers/test-apps/courier/src/main/java/com/twosigma/waiter/courier/GrpcServer.java

```diff
@@ -36,12 +44,81 @@
 public class GrpcServer {
 
     private final static Logger LOGGER = Logger.getLogger(GrpcServer.class.getName());
+    private final static Context.Key<CidTimestamp> CID_TIMESTAMP = Context.key("CID.TIMESTAMP");
+    private final static int MAX_STATES_TRACKED = 100;
+    private final static Map<String, Map<Long, List<String>>> requestCidToStateList;
 
-    private static void sleep(final int durationMillis) {
-        try {
-            Thread.sleep(durationMillis);
-        } catch (final Exception ex) {
-            ex.printStackTrace();
+    static {
```

Comment:
Nitpick: Could we have just kept every CID and avoided the complexity?

Reply:
Done.